includes/Integrations/Jetpack.php

```diff
@@ -184,12 +184,6 @@ public function filter_ajax_query_attachments_args( $args ) {
 				// Load filter at 15, so it load after Media\Media\wp_prepare_attachment_for_js which is loaded at 10.
 				add_filter( 'wp_prepare_attachment_for_js', [ $this, 'filter_admin_ajax_response' ], 15, 2 );
 			}
-
-			$allowed_mime_types_transcodable = array_merge( $allowed_mime_types, $this->types->get_allowed_transcodable_mime_types() );
-			if ( ! array_diff( $allowed_mime_types_transcodable, $args['post_mime_type'] ) ) {
-				// Load filter at 15, so it load after Media\Media\wp_prepare_attachment_for_js which is loaded at 10.
-				add_filter( 'wp_prepare_attachment_for_js', [ $this, 'filter_admin_ajax_response' ], 15, 2 );
-			}
```

Comment:
Hmm I completely forgot about this.

This was added in #7766.

Quote:

> The allow list of transcodable mime type will have to moved to PHP, as it is needed for the jetpack integration.

For context:

Jetpack VideoPress modifies the WP media modal query args quite a bit and forces every VideoPress video to have the mime type `video/videopress` (which is obviously not a real mime type). We want to hijack into that process to disguise them as MP4s (which they are and which is something our editor understands). This is what `filter_admin_ajax_response()` does.
Now, we want to do that at the right time, i.e. only when this media modal is used within the story editor, not interfering with other screens.

This method here tries to do that by being clever, trying to compare the `post_mime_type` query arg with something that the editor might send.

For this section here specifically I am honestly not sure what purpose it serves. I hope @spacedmonkey remembers.

We need to find a way to target Jetpack VideoPress without comparing the list of transcodable mime types.

Reply:
I remember why this code is needed. Consider the following scenario.

- Upload a mov. 
- Videopress converts video to mp4 and uploads to CDN. 
- The attachement's mime type would be video/quicktime. 

This code is needed, to see if the original file was a video file of any kind that has been converted mp4 are usable in editor.

Reply:
Looking at it, if a video has been transcoded to video, why do we care about the orignal video type? 

We could do this. 
```
if ( \in_array( self::VIDEOPRESS_MIME_TYPE, $args['post_mime_type'], true ) ) {
     	add_filter( 'wp_prepare_attachment_for_js', [ $this, 'filter_admin_ajax_response' ], 15, 2 );
}
```